sorts/topological_sort.py

```diff
@@ -8,28 +8,39 @@
 edges = {"a": ["c", "b"], "b": ["d", "e"], "c": [], "d": [], "e": []}
 vertices = ["a", "b", "c", "d", "e"]
 
+# sort stores the topological sort of graph
+sort = []
+# visited dictionary keep tracks of which nodes of graph are visited
+visited = {}
+
 
 def topological_sort(start, visited, sort):
-    """Perform topolical sort on a directed acyclic graph."""
+    """Perform topological sort on a directed acyclic graph."""
     current = start
     # add current to visited
-    visited.append(current)
+    visited[current] = 1
     neighbors = edges[current]
     for neighbor in neighbors:
         # if neighbor not in visited, visit
-        if neighbor not in visited:
+        if visited[neighbor] == 0:
             sort = topological_sort(neighbor, visited, sort)
     # if all neighbors visited add current to sort
     sort.append(current)
-    # if all vertices haven't been visited select a new one to visit
-    if len(visited) != len(vertices):
-        for vertice in vertices:
-            if vertice not in visited:
-                sort = topological_sort(vertice, visited, sort)
     # return sort
     return sort
 
 
 if __name__ == "__main__":
-    sort = topological_sort("a", [], [])
+    # Adding vertices to visited and mark them as 0
+    # 0 value indicates that vertex is unvisited
+    # 1 value indicates that vertex is visited
+    for vertex in vertices:
+        visited.update({vertex: 0})
+
+    for vertex in visited:
+        if visited[vertex] == 0:
+            sort = topological_sort(vertex, visited, sort)
+
+    # reverse the sort
+    sort = sort[::-1]
```

Comment:
Use `doctest` for testing

Reply:
Type hints are added!
I'm a beginner, I'm not aware of doc testing
Thanks for helping me out 😊